usl_pipeline/usl_pipeline/chunkers/elevation_chunkers.py

```diff
@@ -0,0 +1,65 @@
+from osgeo import gdal
+
+from usl_pipeline.readers import elevation_readers
+from usl_pipeline.shared import suppliers
+
+
+class GeoTiffChunker:
+    """Chunker for GeoTIFF elevation data."""
+
+    def __init__(self, elevation_file_path: str, chunk_size: int):
+        """Constructor of GeoTIFF chunker.
+
+        Args:
+            elevation_file_path: GeoTIFF file with elevation data to be split into
+                chunks.
+            chunk_size: Size of a chunk in cells (along both X and Y axes).
+        """
+        self.chunk_size = chunk_size
+        self.elevation_file_path = elevation_file_path
+
+        with open(elevation_file_path, "rb") as input_file:
+            elevation = elevation_readers.read_from_geotiff(
+                input_file, header_only=True
+            )
+
+        self.global_col_count = elevation.header.col_count
+        self.global_row_count = elevation.header.row_count
+        self.x_chunk_count = int((self.global_col_count + chunk_size - 1) / chunk_size)
+        self.y_chunk_count = int((self.global_row_count + chunk_size - 1) / chunk_size)
+
+    def extract_chunk(
+        self, y_chunk_index: int, x_chunk_index: int, chunk_file_path: str
+    ):
+        """Extracts one chunk for a given positional indices.
+
+        Args:
+            y_chunk_index: Y-axis index of the chunk.
+            x_chunk_index: X-axis index of the chunk.
+            chunk_file_path: Output file path to write the chunk to.
+        """
+        row_start = y_chunk_index * self.chunk_size
+        row_count = min(self.global_row_count - row_start, self.chunk_size)
+        col_start = x_chunk_index * self.chunk_size
+        col_count = min(self.global_col_count - col_start, self.chunk_size)
+        ds = gdal.Open(self.elevation_file_path)
+        gdal.Translate(
+            chunk_file_path, ds, srcWin=[col_start, row_start, col_count, row_count]
+        )
+
+    def split_into_chunks(
+        self,
+        chunk_file_path_generator: suppliers.ChunkFilePathGenerator,
```

Comment:
This interface feels like it's exposing some internal concerns to callers.
I'd hope we could get
```python
def split_info_chunks(file: typing.BinaryIO):
```
or 
```python
def split_info_chunks(file_path: str | pathlib.Path):
```
rather than ask callers to need to understand how to work with `suppliers.ChunkFilePathGenerator`. 

`ChunkFilePathGenerator` feels more like an internal interface than a public one to me

Reply:
Now there is prod (default) implementation for this generator so callers now can skip understanding part.